src/xrc-tests/src/tests/basic_exchange_rates.rs

```diff
@@ -0,0 +1,305 @@
+use ic_xrc_types::{Asset, AssetClass, GetExchangeRateRequest, GetExchangeRateResult};
+
+use crate::{
+    container::{run_scenario, Container},
+    mock_responses, ONE_DAY_SECONDS,
+};
+
+/// Setup:
+/// * Deploy mock FOREX data providers and exchanges.
+/// * Start replicas and deploy the XRC, configured to use the mock data sources
+///
+/// Runbook:
+/// * Request exchange rate for various cryptocurrency and fiat currency pairs
+/// * Assert that the returned rates correspond to the expected values
+///
+/// Success criteria:
+/// * All queries return the expected values
+///
+///
+/// How are the expected values determined:
+///
+/// Crypto-pair
+/// 0. The XRC retrieves the ICP/USDT rate.
+///     a. BTC/USDT rates: [ 41960000000, 42030000000, 42640000000, 44250000000, 44833000000, 46022000000, 46101000000, ]
+/// 1. The XRC retrieves the BTC/USDT rate.
+///     a. ICP/USDT rates: [ 41960000000, 42030000000, 42640000000, 44250000000, 44833000000, 46022000000, 46101000000, ]
+/// 2. The XRC divides BTC/USDT by ICP/USDT. The division inverts BTC/USDT to USDT/BTC then multiplies ICP/USDT and USDT/BTC
+///    to get the resulting ICP/BTC rate.
+///     a. ICP/BTC rates: [ 910175465, 911693871, 911737846, 913258858, 924925687, 926513388, 935917716,
+///                         937479065, 948248585, 949830506, 951085114, 959849007, 961496656, 963615816,
+///                         972495153, 974164510, 984052507, 985694158, 986996161, 998286351, 998334516,
+///                         999999974, 999999975, 999999979, 999999980, 999999993, 999999993, 999999997,
+///                         1001668248, 1001716550, 1013175138, 1014513436, 1016205903, 1026520617, 1028282712,
+///                         1037757947, 1040045195, 1041830505, 1051430554, 1052819408, 1054575779, 1066690452,
+///                         1068469964, 1079315169, 1081167889, 1094979769, 1096806474, 1096859379, 1098689220 ]
+/// 3. The XRC returns the median rate and the standard deviation from the BTC/ICP rates.
+///     a. The median rate from step 2 is 999999980.
+///     b. The standard deviation from step 2 is 53827575.
+/// Crypto-fiat pair
+/// 0. The XRC retrieves rates from the mock forex sources.
+///     a. During collection the rates retrieved are normalized to USD.
+///     b. When the collected rates are normalized, then the EUR rate (EUR/USD) is collected (for more information on this collection, see xrc/forex.rs:483).
```

Comment:
Why is the EUR rate collected? The goal is to get the ICP/CXDR rate, right?

Reply:
I'm confused. In the test below, the EUR rate is used. So, why does it say CXDR in step 5?

Reply:
Correct. Fixed the comment.